spec/requests/api/v1/carts/carts_controller_spec.rb

```diff
@@ -0,0 +1,132 @@
+describe '/api/v1/carts/', type: :request do
+  let(:headers) { auth_headers }
+  let(:user) { create(:user) }
+
+  shared_examples 'response bad_request' do
+    it 'returns http bad request code' do
+      expect(response).to be_bad_request
+    end
+  end
+
+  describe 'add_product' do
+    subject { put '/api/v1/carts/add_product', headers: headers, params: params, as: :json }
```

Comment:
use [path helpers](https://guides.rubyonrails.org/routing.html#path-and-url-helpers) here

Reply:
Updated.